pkg/audit/manager.go

```diff
@@ -138,6 +154,100 @@ func (am *Manager) audit(ctx context.Context) error {
 	return am.writeAuditResults(ctx, rs, updateLists, timestamp, totalViolationsPerConstraint)
 }
 
+func (am *Manager) auditResources(ctx context.Context) (*constraintTypes.Responses, error) {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(am.mgr.GetConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	responses := constraintTypes.NewResponses()
+
+	_, serverResourceLists, err := discoveryClient.ServerGroupsAndResources()
+
+	if err != nil {
+		return responses, err
+	}
+
+	errMap := make(opa.ErrorMap)
+
+	clusterAPIResources := make(map[metav1.GroupVersion]map[string]bool)
+	for _, rl := range serverResourceLists {
+		gv := metav1.GroupVersion{
+			Group:   rl.GroupVersionKind().Group,
+			Version: rl.GroupVersionKind().Version,
+		}
+		if _, ok := clusterAPIResources[gv]; !ok {
+			clusterAPIResources[gv] = make(map[string]bool)
+		}
+		for _, resource := range rl.APIResources {
+			if !strings.Contains(resource.Name, "/") {
+				clusterAPIResources[gv][resource.Kind] = true
+			}
+		}
+	}
+
+	for gv, gvKinds := range clusterAPIResources {
+		for kind := range gvKinds {
+			objList := &unstructured.UnstructuredList{}
+			objList.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   gv.Group,
+				Version: gv.Version,
+				Kind:    kind + "List",
+			})
+
+			err := am.client.List(ctx, objList)
+			if err != nil {
+				log.Error(err, "Unable to list objects for gvk", "group", gv.Group, "version", gv.Version, "kind", kind)
+				continue
+			}
+
+			for _, obj := range objList.Items {
+				resourceJSON, err := json.Marshal(obj.Object)
+				if err != nil {
+					log.Error(err, "Unable to marshal JSON encoding of object for audit", "object", obj.Object)
+					continue
+				}
+
+				req := admissionv1beta1.AdmissionRequest{
+					Kind: metav1.GroupVersionKind{
+						Group:   obj.GetObjectKind().GroupVersionKind().Group,
+						Version: obj.GetObjectKind().GroupVersionKind().Version,
+						Kind:    obj.GetObjectKind().GroupVersionKind().Kind,
+					},
+					Object: runtime.RawExtension{
+						Raw: resourceJSON,
+					},
+				}
+
+				resp, err := am.opa.Review(ctx, req)
```

Comment:
We could modify the code for `TargetHandler.Review()` to accept unstructured resources directly.

This would be good because use cases like this are likely to crop up again.

Reply:
Done - added `unstructured.Unstructured` and `*unstructured.Unstructured` support